### ISSUE TITLE
iframe: layers scroll position get lost

### ISSUE THREAD

**Issue by paulrouget** [servo/servo#7864]
Scrolling iframes work, but at the next composition, the layer scroll position get lost. It looks like the page is scrolled up, but it is not (scrolling again re-sync the scroll position).

Here is an example to show what I mean:

```html
<style>
  div {
    width: 100px;
    height: 100px;
    background: red;
  }

  div:hover {
    background: blue;
  }

  iframe {
    margin: 10px;
    width: 600px;
    height: 300px;
  }
</style>

<div></div>
<div></div>

<iframe mozbrowser remote src='http://google.com'></iframe>
```

![colorlesshopefulharvestmouse](https://cloud.githubusercontent.com/assets/373579/10276584/a1b5674e-6b50-11e5-87ae-d4c46ab18e0f.gif)

**Comment by paulrouget** [comment]
Note to self: might want to look at the `self.scroll_layer_to_fragment_point_if_necessary` call in `create_or_update_descendant_layer`.

**Comment by paulrouget** [comment]
Fixed by #8089
